### ISSUE TITLE
address.zipCodeByState generates 4 digits for eastern states

### ISSUE THREAD

**Issue by jbrown123** [Marak/faker.js#1101]
The table for zipcode by state has a min and max for each state. Eastern states in the US (e.g. CT, ME, MA, NH, NJ, NY, RI, VT) have definitions that are only 4 digits (which is correct for the numeric value of their zip range). However, since all we do is pass this to faker.random.number, we end up with 4 digit zipcodes for these states even though US zipcodes are always 5 digits (or 9 if you use Zip+4).

Basically it needs to force a minimum of 5 digits (at least for the US) . Perhaps we need an additional locale-specific definition of `faker.definitions.address.postcode.minlength` that sets a minimum length for zipcodes.

```
this.zipCodeByState = function (state) {
    var zipRange = faker.definitions.address.postcode_by_state[state];
    if (zipRange) {
      return faker.random.number(zipRange);
    }
    return faker.address.zipCode();
  }
```

**Comment by Moosh-be** [comment]
Note "ZIP code" is specific value for the United States and the Philippines.

PostCode is "world name"

- CAP: Italy
- CEP : Brazil
- Eircode: Ireland.
- NPA in French-speaking and Italian-speaking Switzerland.
- PIN: India
- PLZ: Germany, Austria, German-speaking Switzerland and Liechtenstein
- Postal code: Canada.
- Postal index: Eastern European countries such as Ukraine, Moldova, Belarus etc.
- PSČ: Slovakia and Czech Republic

**Comment by jbrown123** [comment]
Yes, it's specific to the US but it still isn't working correctly. By definition, US ZIP codes are 5 digits (or 9 if using ZIP+4). They are NEVER less than 5 digits.

Unfortunately those areas where the ZIP code range is less than 10000 end up generating 4 digit ZIP codes which is incorrect.

**Comment by Marak** [comment]
@jbrown123 -

We will accept a Pull Request to resolve this. 

I suspect we might already have some padding utilities in the library already. If not, we can easily add that to padded us zip codes.

Thank you for letting us know.

**Comment by jbrown123** [comment]
@Marak My inclination would be to add a new property, `postcode_min_length` to the address definitions in lib/index.js. Then populate that in the locales file for the en_US (any other country with a similar problem could use it as well).

Then in lib/address.js add code to this.zipCodeByState to enforce the minimum length, padding with 0's on the left.

My concern with that solution is whether it's the right thing for other countries.

Would you find this an acceptable solution or would you prefer something more "generic"?

**Comment by wolfshirts** [comment]
I have opened a new PR https://github.com/Marak/faker.js/pull/1139 which should address this issue.
